### PR TITLE
add dialog for voting

### PR DESCRIPTION
**Pull request by aguycalled**
Adds a dialog for managing voting of the community fund.

### CODE REVIEW

File: src/qt/forms/moc_cfund_voting.cpp [hunk 1]
```diff
@@ -0,0 +1,145 @@
+/****************************************************************************
+** Meta object code from reading C++ file 'cfund_voting.h'
```
Comment (matt-auckland) [inline src/qt/forms/moc_cfund_voting.cpp:2]: Is this normal?

File: src/qt/forms/moc_cfund_proposals.cpp [hunk 2]
```diff
@@ -0,0 +1,145 @@
+/****************************************************************************
+** Meta object code from reading C++ file 'cfund_proposals.h'
+**
+**
```
Comment (matt-auckland) [inline src/qt/forms/moc_cfund_proposals.cpp]: is this normal?

**Review by matt-auckland — COMMENTED** [review]
Code looks good, will approve after testing

File: src/qt/forms/overviewpage.ui [hunk 3]
```diff
@@ -511,6 +486,26 @@
...
+           <pointsize>11</pointsize>
+           <weight>75</weight>
+           <bold>true</bold>
+          </font>
+         </property>
+         <property name="styleSheet">
+          <string notr="true">color:#cc2222</string>
+         </property>
+         <property name="text">
+          <string>Wallet syncing, the balance won't be accurate until the process finishes</string>
```
Comment (matt-auckland) [inline src/qt/forms/overviewpage.ui]:
```suggestion
          <string>This wallet is currently syncing. Your balance may not be accurate until it has completed</string>
```
Slight wording tweak

**Comment by matt-auckland** [comment]
Shouldn't this be a PR into master? If we want to keep our change atomic like we discussed the other day?

File: src/qt/navcoingui.cpp [hunk 4]
```diff
@@ -420,6 +426,8 @@ void NavCoinGUI::createActions()
     optionsAction->setStatusTip(tr("Modify configuration options for %1").arg(tr(PACKAGE_NAME)));
     optionsAction->setMenuRole(QAction::PreferencesRole);
     optionsAction->setEnabled(false);
+    cfundProposalsAction = new QAction(tr("Voting of Proposals"), this);
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp]:
```suggestion
    cfundProposalsAction = new QAction(tr("Vote for Proposals"), this);
```

File: src/qt/navcoingui.cpp [hunk 5]
```diff
@@ -420,6 +426,8 @@ void NavCoinGUI::createActions()
...
+    cfundPaymentRequestsAction = new QAction(tr("Voting of Payment Requests"), this);
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp]:
```suggestion
    cfundPaymentRequestsAction = new QAction(tr("Vote for Payment Requests"), this);
```

File: src/qt/navcoingui.cpp [hunk 6]
```diff
@@ -1766,6 +1797,62 @@ void NavCoinGUI::updateStakingStatus()
...
+                        continue;
+                    auto it = std::find_if( vAddedPaymentRequestVotes.begin(), vAddedPaymentRequestVotes.end(),
+                        [&prequest](const std::pair<std::string, int>& element){ return element.first == prequest.hash.ToString();} );
+                    if (it == vAddedPaymentRequestVotes.end()) {
+                        fFound = true;
+                        break;
+                    }
+                }
+            }
+            if (fFound && !this->fNotShowAgain && (this->lastDialogShown + (60*60*24)) < GetTimeNow()) {
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp]:
```suggestion
            if (fFound && !this->fDon'tShowAgain && (this->lastDialogShown + (60*60*24)) < GetTimeNow()) {
```

File: src/qt/navcoingui.cpp [hunk 7]
```diff
@@ -1766,6 +1797,62 @@ void NavCoinGUI::updateStakingStatus()
...
+                msgbox.setIcon(QMessageBox::Icon::Warning);
+                msgbox.setCheckBox(cb);
+                QAbstractButton* pButtonInfo = msgbox.addButton(tr("Read about the Community Fund"), QMessageBox::YesRole);
+                QAbstractButton* pButtonOpen = msgbox.addButton(tr("Open Voting Window"), QMessageBox::YesRole);
+                this->lastDialogShown = GetTimeNow();
+
+                msgbox.exec();
+
+                if(cb->isChecked()) {
+                    this->fNotShowAgain = true;
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp]:
```suggestion
                    this->fDon'tShowAgain = true;
```

File: src/qt/navcoingui.cpp [hunk 8]
```diff
@@ -1766,6 +1797,62 @@ void NavCoinGUI::updateStakingStatus()
...
+                } else {
+                    this->fNotShowAgain = false;
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp]:
```suggestion
                    this->fDon'tShowAgain = false;
```

File: src/qt/navcoingui.h [hunk 9]
```diff
@@ -143,6 +145,9 @@ class NavCoinGUI : public QMainWindow
     int prevBlocks;
     int spinnerFrame;
 
+    bool fNotShowAgain;
```
Comment (matt-auckland) [inline src/qt/navcoingui.h]:
```suggestion
    bool fDon'tShowAgain;
```

File: src/qt/navcoingui.cpp [hunk 10]
```diff
@@ -145,6 +149,8 @@ NavCoinGUI::NavCoinGUI(const PlatformStyle *platformStyle, const NetworkStyle *n
     helpMessageDialog(0),
     prevBlocks(0),
     spinnerFrame(0),
+    fNotShowAgain(false),
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp]:
```suggestion
    fDon'tShowAgain(false),
```

just tweaking the wording of this variable a bit

File: src/qt/navcoingui.cpp [hunk 11]
```diff
@@ -277,7 +283,7 @@ NavCoinGUI::NavCoinGUI(const PlatformStyle *platformStyle, const NetworkStyle *n
 
     QTimer *timerStakingIcon = new QTimer(labelStakingIcon);
     connect(timerStakingIcon, SIGNAL(timeout()), this, SLOT(updateStakingStatus()));
-    timerStakingIcon->start(150 * 1000);
+    timerStakingIcon->start(45 * 1000);
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp:286]: Does this mean the wallet checks if we are staking or not even 45 seconds?

File: src/qt/navcoingui.cpp [hunk 12]
```diff
@@ -1766,6 +1797,62 @@ void NavCoinGUI::updateStakingStatus()
...
+                    auto it = std::find_if( vAddedPaymentRequestVotes.begin(), vAddedPaymentRequestVotes.end(),
+                        [&prequest](const std::pair<std::string, int>& element){ return element.first == prequest.hash.ToString();} );
+                    if (it == vAddedPaymentRequestVotes.end()) {
+                        fFound = true;
+                        break;
+                    }
+                }
+            }
+            if (fFound && !this->fNotShowAgain && (this->lastDialogShown + (60*60*24)) < GetTimeNow()) {
+                QCheckBox *cb = new QCheckBox("Do not show this message again during this session.");
```
Comment (matt-auckland) [inline src/qt/navcoingui.cpp]:
```suggestion
                QCheckBox *cb = new QCheckBox("Don't show this notification again until wallet is restarted.");
```

**Review by matt-auckland — APPROVED** [review]

File: src/qt/navcoingui.cpp (hunk 11 above)
Comment (aguycalled) [reply src/qt/navcoingui.cpp:286]: yes

**Pull request closed without merging**